events/tests/test_notifications.py

```diff
@@ -33,6 +34,19 @@
 from users.factories import GuardianFactory
 
 
+def _join_thread_with_name(thread_name: str, ignore_errors=False):
```

Comment:
Maybe the name could be made clearer/easier to understand at a glance? For example `_wait_for_thread` or something similar?